### PR TITLE
Add support for Stack Overflow

### PR DESCRIPTION
**Pull request by kowitz**
#### What's this PR do?
Adds support for StackOverflow suggestions for the following activities:
* Asking a question
* Answering a question

#### Where should the reviewer start?
Feel free to test on stack overflow. I'll send an invite to a private team instance for testing.

#### Any background context you want to provide?
This extension uses a timestamp in an HTML attribute to determine when new activities have happened.

I am unsure of what document type we should use. If anything, this item feels like a `conversation` which is not a document type we currently support. But there are other tools (Intercom, Email) that would use the type `conversation` as well.

#### Definition of Done:

- [ ] Code is easy to understand and conforms with Prettier & eslint configs
- [ ] Incomplete code is marked with TODOs
- [ ] Code is suitably instrumented with logging and metrics
- [ ] Documentation has been updated as appropriate
- [ ] Manifest has been updated and version incremented correctly
- [ ] [OWASP Top 10](https://www.owasp.org/index.php/Top_10-2017_Top_10) have been considered

### CODE REVIEW

File: ext/sites/stackoverflow.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  var sigs = [];
+  // Find the Current UserID from
+  const profile = document.querySelector('.my-profile');
+  const currentUser = reMatch(profile.pathname, reUserId, 1);
+  // Inspect all the question/answer signatures on the page
+  const sigEls = document.querySelectorAll('.post-signature');
+  for (var i = 0; i < sigEls.length; i++) {
+    const userEl = sigEls[i].querySelector('.user-details a');
+    const timeEl = sigEls[i].querySelector('.user-action-time span');
+    if (!userEl || !userEl.pathname || !timeEl || !timeEl.title) break;
```
Comment (chrisforrette) [inline ext/sites/stackoverflow.js]: Should this `break` be a `continue`?

Comment (kowitz) [reply ext/sites/stackoverflow.js]: Yes! Thank you for catching that.

```suggestion
    if (!userEl || !userEl.pathname || !timeEl || !timeEl.title) continue;
``

**Comment by kowitz** [comment]
Oh no, sorry... there was a weird formatting thing that happened when I tried to make the code change via a GitHub suggestion. Just fixed it with that last PR.

**Comment by kowitz** [comment]
@chrisforrette - I think I need another review on this PR. Something about the commit I added dismissed your review.

**Comment by dpup** [comment]
@chrisforrette since this is an open source repository where external contributors can send PRs I wanted to make sure that followup commits required a review.

**Comment by chrisforrette** [comment]
@dpup Ah, that makes sense.

**Pull request merged**